### PR TITLE
Updated activity schedule

### LINKED ISSUE #1455: Replace date picker and use review pattern for milestones in Activity Schedule subsection (Program Activities subsection), and make consistent with design system
Within the Program Activities section, the 3rd subsection is Activity Schedule. This subsection currently uses the calendar picker, but there is no calendar component in the CMS design system and they recommend using the date field instead. The date field here is using the blue line pattern from the checkboxes/radio buttons with children.
image
image

### This task is done when...
 - [x] `Activity Schedule` is `h4.ds-h4`
 - [ ] `Activity start and end dates` and `Milestones` labels are `h5.ds-h5`
 - [ ] Date entry boxes have been swapped out for a new multi-box format
 - [ ] List of milestones uses the review pattern for adding, editing, and removing milestones
 - [ ] Milestone titles are `h6.ds-h4` (semantically smaller but visually larger)
 - [ ] There is an `hr` at the bottom of the section

### PR DESCRIPTION
**Pull request by mgwalker**
Describe the pull request here, including any supplemental information needed to understand it.

### This pull request changes...
- switches to design system form elements
- wraps design system date field in one that automatically parses and returns the date format we use internally, so each component doesn't have to do that
- switches to new date field
- adds review pattern
- closes #1455 

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This feature is done when...
- [ ] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

**Comment by mgwalker** [comment]
100% in-scope, my mistake for getting it wrong. 😬

File: web/src/containers/activity/Milestone.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import Form from './MilestoneForm';
+import Review from './MilestoneReview';
+
+const Milestone = ({ initialCollapsed, ...rest }) => {
+  const [collapsed, setCollapsed] = useState(initialCollapsed);
+  const collapse = useCallback(() => setCollapsed(true), []);
+  const expand = useCallback(() => setCollapsed(false), []);
+
+  if (collapsed) {
+    return <Review {...rest} expand={expand} />;
```
Comment (hbillings) [inline web/src/containers/activity/Milestone.js]: Having this `Review` pattern here is just so :chef_kiss:

File: web/src/containers/activity/Schedule.js [hunk 2]
```diff
@@ -1,136 +1,104 @@
...
+  const handleMilestoneNameChange = useCallback((index, name) => {
+    updateActivity(activity.key, {
+      schedule: { [index]: { milestone: name } }
+    });
+  });
 
-    const updates = { schedule: { [index]: { endDate } } };
-    updateActivity(activity.key, updates);
-  };
+  const handleMilestoneDateChange = useCallback((index, date) => {
```
Comment (hbillings) [inline web/src/containers/activity/Schedule.js]: I really like how much more explicit this is.

File: web/package.json [hunk 3]
```diff
@@ -110,8 +110,8 @@
     "css-loader": "^2.1.0",
     "cssnano": "^4.1.10",
     "d3-geo": "^1.11.1",
-    "enzyme": "^3.9.0",
-    "enzyme-adapter-react-16": "^1.10.0",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.14.0",
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/package.json]: Better support for hooks in the `Shallow` renderer.  It still doesn't seem to support state hooks that change rendering behavior so there's still a bit of hacking necessary to test those kinds of components, but this made most of the tests a lot simpler.

**Comment by mgwalker** [comment]
@jeromeleecms Oops, responded a little too fast.  The second part, the date format, is in scope.  The other one I think is out of scope because the date field is from the design system.  We're not actually controlling the `<input>` elements that get rendered there, so we can't set configure them unless the design system gives us a way to do that - and in this case, it does not.  That seems like a really worthwhile issue to file with them, though.

File: web/src/containers/activity/MilestoneForm.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+const MilestoneForm = ({
+  idx,
+  endDate,
+  onChangeDate,
+  onChangeName,
+  name,
+  collapse
+}) => {
+  const changeDate = useCallback(
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/containers/activity/MilestoneForm.js]: Maybe I should submit a tech talk on React hooks? They're pretty nifty.

**Pull request merged**